packages/fx-core/src/component/generator/utils.ts

```diff
@@ -117,6 +117,15 @@ export async function fetchTemplateZipUrl(
   if (!selectedTag) {
     throw new Error(`Failed to find valid template for ${name}`);
   }
+  return selectedTag;
+}
+
+export async function fetchTemplateZipUrl(
+  name: string,
+  tryLimits = defaultTryLimits,
+  timeoutInMs = defaultTimeoutInMs
+): Promise<string> {
+  const selectedTag = await getTemplateLatestVersion(name, tryLimits, timeoutInMs);
```

Comment:
This function sends two HTTP requests in one call. It's better to avoid calling this function twice.

Reply:
I will optimize this function.